Code/Mantid/Framework/DataHandling/src/CheckMantidVersion.cpp

```diff
@@ -0,0 +1,234 @@
+#include "MantidDataHandling/CheckMantidVersion.h"
+#include "MantidKernel/InternetHelper.h"
+#include "MantidKernel/MantidVersion.h"
+#include "MantidKernel/Strings.h"
+
+#include <Poco/DateTimeFormatter.h>
+#include <Poco/DateTimeFormat.h>
+#include <Poco/DateTimeParser.h>
+#include <Poco/StringTokenizer.h>
+
+// jsoncpp
+#include <json/json.h>
+
+#include <boost/lexical_cast.hpp>
+
+namespace Mantid {
+namespace DataHandling {
+
+using namespace Mantid::Kernel;
+using Mantid::API::WorkspaceProperty;
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(CheckMantidVersion)
+
+//----------------------------------------------------------------------------------------------
+/** Constructor
+ */
+CheckMantidVersion::CheckMantidVersion() {}
+
+//----------------------------------------------------------------------------------------------
+/** Destructor
+ */
+CheckMantidVersion::~CheckMantidVersion() {}
+
+//----------------------------------------------------------------------------------------------
+
+/// Algorithms name for identification. @see Algorithm::name
+const std::string CheckMantidVersion::name() const {
+  return "CheckMantidVersion";
+}
+
+/// Algorithm's version for identification. @see Algorithm::version
+int CheckMantidVersion::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+const std::string CheckMantidVersion::category() const {
+  return "Utility\\Development";
+}
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+const std::string CheckMantidVersion::summary() const {
+  return "Checks if there is a more recent version of Mantid available using "
+         "the Github API";
+}
+
+//----------------------------------------------------------------------------------------------
+/** Initialize the algorithm's properties.
+ */
+void CheckMantidVersion::init() {
+  declareProperty("CurrentVersion", "", Direction::Output);
+  declareProperty("MostRecentVersion", "", Direction::Output);
+  declareProperty("IsNewVersionAvailable", false,"True if a newer version is available, otherwise false", Direction::Output);
+}
+
+  //----------------------------------------------------------------------------------------------
+/** Execute the algorithm.
+ */
+void CheckMantidVersion::exec() {
+  std::string currentVersion = getCurrentVersion();
+  setProperty("CurrentVersion", currentVersion);
+  std::string mostRecentVersion = "";
+
+  std::string gitHubReleaseUrl = ConfigService::Instance().getString(
+      "CheckMantidVersion.GitHubReleaseURL");
+  if (gitHubReleaseUrl.empty()) {
+    gitHubReleaseUrl =
+        "https://api.github.com/repos/mantidproject/mantid/releases/latest";
+  }
```

Comment:
When I see this I somehow feel that these defaults should be the responsibility of the ConfigService?
Not really related to this PR.